internal/processmon/processmon.go

```diff
@@ -0,0 +1,379 @@
+	rpcClientSecret := constants.AporetoEnvRPCClientSecret + "=" + randomkeystring
+	envStatsSecret := constants.AporetoEnvStatsSecret + "=" + statsServerSecret
+	containerPID := constants.AporetoEnvContainerPID + "=" + strconv.Itoa(refPid)
+
+	newEnvVars := []string{
+		mountPoint,
+		namedPipe,
+		statsChannel,
+		rpcClientSecret,
+		envStatsSecret,
+		containerPID,
+	}
+
+	// If the PURuntime Specified a NSPath, then it is added as a new env var also.
+	if refNSPath != "" {
+		nsPath := constants.AporetoEnvNSPath + "=" + refNSPath
+		newEnvVars = append(newEnvVars, nsPath)
+	}
+
+	return newEnvVars
+}
+
+// LaunchProcess prepares the environment and launches the process
+func (p *processMon) LaunchProcess(
+	contextID string,
+	refPid int,
+	refNSPath string,
+	rpchdl rpcwrapper.RPCClient,
+	arg string,
+	statsServerSecret string,
+	procMountPoint string,
+) error {
+
+	_, err := p.activeProcesses.Get(contextID)
+	if err == nil {
+		return nil
+	}
+
+	var nsPath string
+
+	// We check if the NetNsPath was given as parameter.
+	// If it was we will use it. Otherwise we will determine it based on the PID.
+	if refNSPath == "" {
+		nsPath = filepath.Join(procMountPoint, strconv.Itoa(refPid), "ns/net")
+	} else {
+		nsPath = refNSPath
+	}
+
+	hoststat, hoststaterr := os.Stat(filepath.Join(procMountPoint, "1/ns/net"))
+	if hoststaterr != nil {
+		return hoststaterr
+	}
+
+	pidstat, pidstaterr := os.Stat(nsPath)
+	if pidstaterr != nil {
+		return fmt.Errorf("Container pid not found: %d %s", refPid, pidstaterr.Error())
+	}
+
+	if pidstat.Sys().(*syscall.Stat_t).Ino == hoststat.Sys().(*syscall.Stat_t).Ino {
+		return fmt.Errorf("Refused to launch a remote enforcer in host namespace")
+	}
+
+	if _, staterr := os.Stat(p.netNSPath); staterr != nil {
+		mkerr := os.MkdirAll(p.netNSPath, os.ModeDir)
+		if mkerr != nil {
+			zap.L().Warn("Could not create directory", zap.Error(mkerr))
+		}
+	}
+
+	// A symlink is created from /var/run/netns/<context> to the NetNSPath
+	contextFile := filepath.Join(p.netNSPath, contextID)
+	if _, lerr := os.Stat(contextFile); lerr != nil {
+		if linkErr := os.Symlink(nsPath, contextFile); linkErr != nil {
+			zap.L().Warn("Failed to create symlink for use by ip netns", zap.Error(linkErr))
+		}
+	}
+
+	cmd, err := p.getLaunchProcessCmd(arg, contextID)
+	if err != nil {
+		return fmt.Errorf("Enforcer Binary not found: %s", err.Error())
+	}
+
+	exited := make(chan int, 2)
+	waitForExitCount := 0
+	if p.logToConsole {
+		if waitForExitCount, err = p.pollStdOutAndErr(cmd, exited, contextID); err != nil {
+			return err
+		}
+	}
+
+	randomkeystring, err := crypto.GenerateRandomString(secretLength)
+	if err != nil {
+		//This is a more serious failure. We can't reliably control the remote enforcer
+		return fmt.Errorf("Failed to generate secret: %s", err.Error())
+	}
+
+	// Start command
+	newEnvVars := p.getLaunchProcessEnvVars(
+		procMountPoint,
+		contextID,
+		randomkeystring,
+		statsServerSecret,
+		refPid,
+		refNSPath)
```

Comment:
```go
newEnvVars := p.getLaunchProcessEnvVars(
 		procMountPoint,
 		contextID,
 		randomkeystring,
 		statsServerSecret,
 		refPid,
 		refNSPath,
)
```

Reply:
how much to indent ? This is done by editor setup.

Reply:
let the indent done by the  goformat, it's just the last argument should end up with `,\n)`, not `)`